src/app/listing-toolbar/listing-toolbar.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, ViewEncapsulation, EventEmitter, Output } from '@angular/core';
+
+const SORT_OBJECTS = [
+  { text: 'Price: Low to High', params: { sort: 'price', direction: 'ascending' } },
+  { text: 'Price: High to Low', params: { sort: 'price', direction: 'descending' } },
+  { text: 'Seller Rating', params: { sort: 'userRating', direction: 'descending' } },
+  { text: 'Date: Most Recent', params: { sort: 'createdAt', direction: 'descending' } },
+  { text: 'Date: Least Recent', params: { sort: 'createdAt', direction: 'ascending' } }
+];
+
+const CATEGORY_OBJECTS = [
+  { text: 'All', value: 'all' }
+];
+
+@Component({
+  selector: 'app-listing-toolbar',
+  templateUrl: './listing-toolbar.component.html',
+  styleUrls: ['./listing-toolbar.component.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class ListingToolbarComponent {
+
+  @Output() listEvent = new EventEmitter<string>();
+
+  private categoryObjects: Object[] = CATEGORY_OBJECTS;
+  private selectedCategory: string;
+  private sortObjects: Object[] = SORT_OBJECTS;
+  private selectedSort: Object;
```

Comment:
In TypeScript, objects are actually a primitive type. Use `object` instead of `Object`, otherwise TypeScript may not recognize certain `object` methods if you use them.